arangod/Cluster/DropCollection.cpp

```diff
@@ -109,6 +110,8 @@ bool DropCollection::first() {
 
   notify();
 
+  _feature.delShardVersion(collection);
```

Comment:
This feels like it is creating a race condition, not preventing one.  Especially since it is after the notify() call instead of before.  What is the sequence of Create, Sync, Drop actions that is being addressed throughout this PR.  Documentation and comments are zero.

Reply:
Agreed. Fixed.